### PR TITLE
Implements devise token auth, opens up app for API access

### PR DESCRIPTION
**Pull request by CamronLDNF**
# Description
* Opens up the app for API access from external clients, such as our mobile client, to access articles.
* Sets up request specs

# Pivotal Tracker
* [Backend API access](https://www.pivotaltracker.com/story/show/163343938)
* [Test environment](https://www.pivotaltracker.com/story/show/163343961)

# Gems added
* `rack-cors`
* `devise_token_auth`

# Screenshot
image

### CODE REVIEW

**Review by johannaanden — APPROVED** [review]
LGTM

File: spec/requests/api/registrations_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      FactoryBot.create(:user, email: 'me@mail.com',
+                               password: 'whatever',
+                               password_confirmation: 'whatever')
+
+      post '/api/auth', params: {email: 'me@mail.com',
+                                 password: 'whatever',
+                                 password_confirmation: 'whatever'
+                              }, headers: headers
+
+      expect(JSON.parse(response.body)['errors']['email']).to eq ['has already been taken']
```
Comment (Jon1969Edwards) [inline spec/requests/api/registrations_spec.rb:49]: Should this:  ['has already been taken']  : be capitalized?

Comment (CamronLDNF) [reply spec/requests/api/registrations_spec.rb:49]: The json response returns in lower case, so this is correct. All tests are going green.

**Review by tochman — APPROVED** [review]
Looks good. Please create a ref ticket for the JSON.parse repetition.

**Pull request merged**